server/src/main/java/com/cloud/storage/StorageManagerImpl.java

```diff
@@ -460,50 +463,37 @@ public Answer sendToPool(StoragePool pool, Command cmd) throws StorageUnavailabl
         return answers[0];
     }
 
-    private GetStorageStatsAnswer getStoragePoolStats(StoragePool pool, GetStorageStatsCommand cmd) {
-        DataStoreProvider storeProvider = _dataStoreProviderMgr.getDataStoreProvider(pool.getStorageProviderName());
-        DataStoreDriver storeDriver = storeProvider.getDataStoreDriver();
+    private GetStorageStatsAnswer getStoragePoolStats(StoragePool pool, GetStorageStatsCommand cmd, DataStoreDriver storeDriver) {
```

Comment:
```suggestion
    private GetStorageStatsAnswer getStoragePoolStats(StoragePool pool, GetStorageStatsCommand cmd) {
```

@slavkap can you pick `storeDriver` from the pool itself, instead of explicitly pass as argument here.

```
DataStoreProvider storeProvider = _dataStoreProviderMgr.getDataStoreProvider(pool.getStorageProviderName());
DataStoreDriver storeDriver = storeProvider.getDataStoreDriver();
```